pkg/virt-api/rest/interfacehotplug_test.go

```diff
@@ -0,0 +1,344 @@
+
+			mockScenario(addOpts)
+			Expect(response.StatusCode()).To(Equal(http.StatusAccepted))
+		},
+			Entry("VM with a valid add interface request", &v1.AddInterfaceOptions{
+				NetworkName:   networkToHotplug,
+				InterfaceName: ifaceToHotplug,
+			}, successfulMockScenarioForVM),
+			Entry("VMI with a valid add interface request", &v1.AddInterfaceOptions{
+				NetworkName:   networkToHotplug,
+				InterfaceName: ifaceToHotplug,
+			}, successfulMockScenarioForVMI),
+		)
+
+		DescribeTable("Should fail on invalid requests for a dynamic interfaces", func(addOpts *v1.AddInterfaceOptions, mockScenario func(addOpts *v1.AddInterfaceOptions), featuresToEnable ...string) {
+			for _, featureToEnable := range featuresToEnable {
+				enableFeatureGate(featureToEnable)
+			}
+
+			mockScenario(addOpts)
+			Expect(response.StatusCode()).To(Equal(http.StatusBadRequest))
+		},
+			Entry("VM with an invalid add interface request missing a network name", &v1.AddInterfaceOptions{
+				InterfaceName: ifaceToHotplug,
+			}, failedMockScenarioForVM, virtconfig.HotplugNetworkIfacesGate),
+			Entry("VM with an invalid add interface request missing the interface name", &v1.AddInterfaceOptions{
+				NetworkName: networkToHotplug,
+			}, failedMockScenarioForVM, virtconfig.HotplugNetworkIfacesGate),
+			Entry("VM with a valid add interface request but no feature gate", &v1.AddInterfaceOptions{
+				NetworkName:   networkToHotplug,
+				InterfaceName: ifaceToHotplug,
+			}, failedMockScenarioForVM),
+		)
+
+		DescribeTable("Should generate expected vmi patch", func(interfaceRequest *v1.VirtualMachineInterfaceRequest, expectedPatch string) {
+			vmi := api.NewMinimalVMI(request.PathParameter("name"))
+			vmi.Namespace = "default"
+			vmi.Status.Phase = v1.Running
+			vmi.Spec.Domain.Devices.Interfaces = append(vmi.Spec.Domain.Devices.Interfaces, v1.Interface{
+				Name: existingIfaceName,
+			})
+			vmi.Spec.Networks = append(vmi.Spec.Networks, v1.Network{
+				Name: existingIfaceName,
+			})
+
+			patch, err := generateVMIInterfaceRequestPatch(vmi, interfaceRequest)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(patch).To(Equal(expectedPatch))
+		},
+			Entry("when the add interface request features all required data",
+				&v1.VirtualMachineInterfaceRequest{
+					AddInterfaceOptions: &v1.AddInterfaceOptions{
+						NetworkName:   networkToHotplug,
+						InterfaceName: ifaceToHotplug,
+					},
+				},
+				fmt.Sprintf("[{ \"op\": \"test\", \"path\": \"/spec/networks\", \"value\": [{\"name\":\"%[1]s\"}]}, { \"op\": \"test\", \"path\": \"/spec/domain/devices/interfaces\", \"value\": [{\"name\":\"%[1]s\"}]}, { \"op\": \"replace\", \"path\": \"/spec/networks\", \"value\": [{\"name\":\"%[1]s\"},{\"name\":\"%[2]s\",\"multus\":{\"networkName\":\"%[3]s\"}}]}, { \"op\": \"replace\", \"path\": \"/spec/domain/devices/interfaces\", \"value\": [{\"name\":\"%[1]s\"},{\"name\":\"%[2]s\",\"bridge\":{}}]}]", existingIfaceName, ifaceToHotplug, networkToHotplug)),
```

Comment:
It is extremely challenging to read this jsom dumps.
Can we normalize it in some way so it will be readable? (e.g. having newlines and not needing the escape character).